docs/usage-tui.md

```diff
@@ -0,0 +1,207 @@
+
+You can also specify some of the command's optional arguments without specifying others. For
+example, `new-wallet` command creates a new wallet and expects three arguments: `pass`, `name` and
+`entropy-size`. You can either call it with both passphrase and name, omitting argument names:
+
+    new-wallet "cat" "test"
+
+Or, if you want to use the default (empty) passphrase, but need to give the wallet a name, you can
+specify only the second argument:
+
+    new-wallet name: "test"
+
+`Knit` language has literals of several formats:
+
+- Strings in double quotes (`"string"`)
+- Numbers with or without decimal point (`1`, `0.5`) or in scientific form (`1e-6`)
+- ADA coin amounts (`1.5ADA`, see below)
+- Cardano addresses without quotes (`sxtitePxjp5dcfm1u8gWgDBGMCEZMhGa6kUPu8VHhqpCtBDPExrJTTCCUHKkyEJSgjb41JT5Tfh1QXb7uUpgjyBKMw`)
+- Task ids (`<1>`, see below)
+- File paths, either relative or absolute, without quotes (`./file.txt`, `/etc/file.txt`)
+
+Let's first cover `Knit` commands that are not related to cryptocurrency. There are not many:
+
+- `L`: create a list of things. Call it with any number of arguments, which may be of different
+    types:
+
+ ```
+ L 1 2 3 "a" "b" "c"
+ ```
+
+- `help` and `logs`: switch to Help and Logs tabs respectively
+- `print`: print an argument of any type
+- `sleep`: delay execution for a given number of microseconds
+- `not`, `true` and `false`: boolean negation and two boolean literals
+
+Aside from that you can enter multiple commands in one input. To do so, separate the commands with
+`;`.  REPL widget input line also allows you to break lines for visual stylization of commands. If
+you enter a backslash (`\ `) and press the Enter key, a new input line will appear.
+
+All `Knit` commands are executed asynchronously, giving you back the control of REPL right after you
+press Enter. Ariadne has a task manager for these background commands.  When you run some
+long-running task, its id (a number in angle brackets like `<1>`) and a string `"Waiting for
+result"` will appear. To cancel a command, use `kill` command like so:
+
+    kill <1>
+
+You can also make some commands depend on others by using `wait` command with a specific task id.
+
+### User interface and Knit integration
+
+Some of the `Knit` functions may access the internal state of Ariadne UI &mdash; namely, currently
+selected wallet, account or address. This allows you to omit the wallet name when creating a new
+account using `new-account` command &mdash; it will use the currently selected one. There is also a
+`copy-selection` command, which will copy the name of selected item (wallet, account or address) to
+the system clipboard.
+
+### Cardano-related commands
+
+Finally, to the commands which will help you manage your `ADA` wallet. Each command will be
+annotated with its arguments and their types. Optional arguments will be marked with "`?`".
+
+- `new-wallet pass: String? name: String? entropy-size: Int?`: create a new wallet with a
```

Comment:
I'd mention that we use the approach from BIP-39.

Reply:
Ok.